tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc

```diff
@@ -38,6 +41,9 @@ namespace {
 
 class LegalizeTFNoFallback
     : public LegalizeTFNoFallbackBase<LegalizeTFNoFallback> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<scf::SCFDialect>();
```

Comment:
Is this needed? The `LegalizeTFNoFallback` pass will not create any scf ops, will it?

Reply:
I think unless I added it the code doesn't allow me to legalize YieldOp and IfOp.